activity.php

```diff
@@ -19,37 +22,31 @@ public function printData(){
         }
 
         // define contracts for basic fetch and mutate functionality
-        abstract public function getActivities() : iterable;
-        abstract public function getActivity($key);
-        abstract public function addActivity($key, $value);
-        abstract public function deleteActivity($key): iterable;
+        abstract public function get() : Ds\Sequence;
+        abstract public function add($key, $value);
+        abstract public function delete($key) : bool;
     }
 
     class ActivityImpl extends Activity {
-        public function getActivities(): iterable {
-            return $this->data;
+        public function get() : Ds\Sequence {
+            return $this->data->pairs();
         }
 
-        public function getActivity($key) {
-            return $this->data[$key];
-        }
-
-        public function addActivity($key, $value) {
+        public function add($key, $value) {
             $this->data[$key] = $value;
-            return $this->data;
+            return $this->data[$key];
         }
 
-        public function deleteActivity($key): iterable {
+        public function delete($key) : bool {
             unset($this->data[$key]);
-            return $this->data;
+            return $this->data->hasKey($key) ? false : true;
```

Comment:
Return false if key was not found

Reply:
What you want to do here is assume remove operation will pass if key exists however if key does not already exist return false

Reply:
I added a try/catch to return false when the key isn't present in the map.